tests/integration_tests/test_xgboost.py

```diff
@@ -1,48 +1,73 @@
 import xgboost as xgb
 
 import optuna
+from optuna.integration.xgboost import use_callback_cls
 from optuna.integration.xgboost import XGBoostPruningCallback
 from optuna.testing.integration import create_running_trial
 from optuna.testing.integration import DeterministicPruner
 
 
 def test_xgboost_pruning_callback_call() -> None:
+    if use_callback_cls is False:
+        env = xgb.core.CallbackEnv(
+            model="test",
+            cvfolds=1,
+            begin_iteration=0,
+            end_iteration=1,
+            rank=1,
+            iteration=1,
+            evaluation_result_list=[["validation-error", 1.0]],
+        )
 
-    env = xgb.core.CallbackEnv(
-        model="test",
-        cvfolds=1,
-        begin_iteration=0,
-        end_iteration=1,
-        rank=1,
-        iteration=1,
-        evaluation_result_list=[["validation-error", 1.0]],
-    )
-
-    # The pruner is deactivated.
-    study = optuna.create_study(pruner=DeterministicPruner(False))
-    trial = create_running_trial(study, 1.0)
-    pruning_callback = XGBoostPruningCallback(trial, "validation-error")
-    pruning_callback(env)
-
-    # The pruner is activated.
-    study = optuna.create_study(pruner=DeterministicPruner(True))
-    trial = create_running_trial(study, 1.0)
-    pruning_callback = XGBoostPruningCallback(trial, "validation-error")
-    with pytest.raises(optuna.TrialPruned):
+        # The pruner is deactivated.
+        study = optuna.create_study(pruner=DeterministicPruner(False))
+        trial = create_running_trial(study, 1.0)
+        pruning_callback = XGBoostPruningCallback(trial, "validation-error")
         pruning_callback(env)
 
+        # The pruner is activated.
+        study = optuna.create_study(pruner=DeterministicPruner(True))
+        trial = create_running_trial(study, 1.0)
+        pruning_callback = XGBoostPruningCallback(trial, "validation-error")
+        with pytest.raises(optuna.TrialPruned):
+            pruning_callback(env)
+    else:
+        # The pruner is deactivated.
+        study = optuna.create_study(pruner=DeterministicPruner(False))
+        trial = create_running_trial(study, 1.0)
+        pruning_callback = XGBoostPruningCallback(trial, "validation-logloss")
+        pruning_callback.after_iteration(
+            model=None, epoch=1, evals_log={"validation": OrderedDict({"logloss": [1.0]})}
+        )
+
+        # The pruner is activated.
+        study = optuna.create_study(pruner=DeterministicPruner(True))
+        trial = create_running_trial(study, 1.0)
+        pruning_callback = XGBoostPruningCallback(trial, "validation-logloss")
+        with pytest.raises(optuna.TrialPruned):
+            pruning_callback.after_iteration(
+                model=None, epoch=1, evals_log={"validation": OrderedDict({"logloss": [1.0]})}
+            )
+
 
 def test_xgboost_pruning_callback() -> None:
+    if use_callback_cls:
+        key = "validation-logloss"
+    else:
+        key = "validation-error"
```

Comment:
repeating "Hmm, I think we don't need this block because I guess the CI will install xgboost>=1.3 hereafter and this block will not be tested."